app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
```

Comment:
>  I could just set @significant_status = nil instead of the reload.

:+1: you can reset the variable to nil right after you create the status. Or just remove the memoization, but that might generate a bit too much queries.